### PR TITLE
Add react router sample

### PR DESCRIPTION
**Pull request by dustinsoftware**

### CODE REVIEW

**Comment by dustinsoftware** [comment]
Fixes #476

File: src/React.Router/HtmlHelperExtensions.cs [hunk 1]
```diff
@@ -123,7 +123,7 @@ var reactComponent
 
 				var executionResult = reactComponent.RenderRouterWithContext(clientOnly, serverOnly);
 
-				if (executionResult.Context?.status != null)
+				if (executionResult.Context?.status != null || executionResult.Context?.url != null)
```
Comment (Daniel15) [inline src/React.Router/HtmlHelperExtensions.cs]: Why is this change needed?

File: src/React.Router/SetServerResponse.cs [hunk 2]
```diff
@@ -36,7 +36,7 @@ public static class SetServerResponse
 		/// <param name="Response">The response object to use.</param>
 		public static void ModifyResponse(RoutingContext context, HttpResponse Response)
 		{
-			var statusCode = context.status.Value;
+			var statusCode = context.status ?? 302;
```
Comment (Daniel15) [inline src/React.Router/SetServerResponse.cs]: Why is this change needed?

File: src/React.Router/HtmlHelperExtensions.cs (hunk 1 above)
Comment (dustinsoftware) [reply src/React.Router/HtmlHelperExtensions.cs]: It is possible for a redirect to be returned [without a status code defined](https://github.com/ReactTraining/react-router/blob/master/packages/react-router/docs/api/Redirect.md)

File: src/React.Router/SetServerResponse.cs (hunk 2 above)
Comment (dustinsoftware) [reply src/React.Router/SetServerResponse.cs]: If a status code is omitted we can't know if the author intended a permanent or temporary redirect, therefore it's safest to assume temporary

**Comment by Daniel15** [comment]
Thanks!
